File: DataStructures/Set/AVLSet.js
const AVLTree = require('../AVLTree/AVLTree')

/**
 * @name AVLSet 集合
 * @description 使用 AVLTree 实现集合
 */
class AVLSet {
  constructor() {
    this.avl = new AVLTree()
  }

  getSize() {
    return this.avl.getSize()
  }

  isEmpty() {
    return this.avl.isEmpty()
  }

  contains(element) {
    return this.avl.contains(element)
  }

  add(element) {
    this.avl.add(element, null)
  }

  remove(element) {
    this.avl.remove(element)
  }
}

module.exports = AVLSet

File: Java/DataStructures/Set/AVLSet.java
/**
 * AVLSet 使用 AVLTree 实现 Set
 */
public class AVLSet<E extends Comparable<E>> implements Set<E> {

  private AVLTree<E, Object> avl;

  public AVLSet() {
    avl = new AVLTree<>();
  }

  @Override
  public void add(E e) {
    avl.add(e, null); // value 使用 null 代替
  }

  @Override
  public void remove(E e) {
    avl.remove(e);
  }

  @Override
  public boolean contains(E e) {
    return avl.contains(e);
  }

  @Override
  public int getSize() {
    return avl.getSize();
  }

  @Override
  public boolean isEmpty() {
    return avl.isEmpty();
  }
}